static-site/src/sections/edit-group-settings-page.jsx

```diff
@@ -0,0 +1,86 @@
+import React, { useEffect, useState } from "react";
+import * as splashStyles from "../components/splash/styles";
+import { ErrorBanner } from "../components/splash/errorMessages";
+import GenericPage from "../layouts/generic-page";
+import { FlexGrid, FlexGridRight, HugeSpacer, MediumSpacer } from "../layouts/generalComponents";
+import EditLogoForm from "../components/Groups/edit-logo-form";
+import EditOverviewForm from "../components/Groups/edit-overview-form";
+
+const UNAUTHORIZED_MESSAGE = `
+  You must have the owners role within a group to edit group settings.
+  If your permissions have changed recently, try logging out and logging back in.
+`;
+
+const EditGroupSettingsPage = ({ location, groupName }) => {
+  const [ userAuthorized, setUserAuthorized ] = useState(true);
```

Comment:
The way the `return` is written and the default of `userAuthorized=true` here means that, for an unauthorized user, the editor is shown until `checkUserAuthz()` finishes running.

Suggestions:

1. Default to `userAuthorized=null` and return some simple "loading" text while it is still that value.
2. After `checkUserAuthz()`, `userAuthorized` will be either `true` or `false`. Based on the value, return the editor or the error banner.

Reply:
Updated this in d02f396a547f2802c4814453db0f1ab394419a46